### ISSUE TITLE
context-root for web-ext is no longer honored with WLP 22.0.0.1

### ISSUE THREAD

**Issue by nickolas-potvin** [OpenLiberty/open-liberty#19937]
Specifying the the `context-root` in a `web-ext` element, e.g:
```
<web-ext id="namehere" moduleName="namehere" context-root="/" />
```

is now ignored starting with WLP 22.0.0.1. This works with WLP 21.0.0.12 as expected.

The context root for the app in the above example becomes `/namehere` instead of `/`.

**Comment by jhanders34** [comment]
Based off of the description, this should be treated as a #regression introduced in 22.0.0.1

**Comment by cbridgha** [comment]
@jimblye any chance some of the changes you made here are related?  https://github.com/OpenLiberty/open-liberty/pull/17303

**Comment by nickolas-potvin** [comment]
I'm not sure if this is helpful but this behavior also shows up in `messages.log` entries. Example for:
```
<web-ext id="com.ibm.test.rest" moduleName="com.ibm.test.rest" context-root="restapi" />
```
21.0.0.12:
```
SRVE0242I: [<server>] [/restapi] [api-rest]: Initialization successful.
```
22.0.0.1:
```
SRVE0242I: [<server>] [/com.ibm.test.rest] [api-rest]: Initialization successful.
```

**Comment by brenthdaniel** [comment]
The issue was caused by #11911  .. In particular, the new code will only look for module info in the non-persistent cache when the parent overlay is not null:
```
    if ( (appInfo == null) && (ddOverlay.getParentOverlay() != null) ) {
            moduleInfo = (ModuleInfo)
                ddOverlay.getFromNonPersistentCache(cachePath, ModuleInfo.class);
```

I'm not really sure why this check was added, but in the particular case I tested (EAR with an application.xml, app expand on) the parent overlay was null but the module information was in the cache. It looks to me like `ddOverlay` is the overlay file for the ear, but this code is expecting it to be for the war.

**Comment by tbitonti** [comment]
Trying to reproduce locally, but so far haven't been able to.  Asked for trace com.ibm.ws.javaee.ddmodel*=finer.
Continuing to attempt to reproduce locally.

**Comment by tbitonti** [comment]
Reproduced locally.  The failure occurs when autoExpand is true.  This addition caused the failure to occur:
`    <applicationManager autoExpand="true"/>`

**Comment by tbitonti** [comment]
The check is a generalization of the check done for EJB bindings.  Contrast these two earlier versions of bnd-ext processing:
ejb-bnd:
https://github.com/OpenLiberty/open-liberty/blob/b7706872be0ae205276922229b7793a22eb96b3b/dev/com.ibm.ws.javaee.ddmodel/src.gen/com/ibm/ws/javaee/ddmodel/ejbbnd/EJBJarBndAdapter.java
web-ext:
https://github.com/OpenLiberty/open-liberty/blob/a692d00deb61d2363d5eab9b0d50441062be13f8/dev/com.ibm.ws.javaee.ddmodel/src.gen/com/ibm/ws/javaee/ddmodel/webext/WebExtAdapter.java

**Comment by tbitonti** [comment]
My take-away is that there are likely similar problems with EJB bindings and extensions configuration overrides.

For now, I'll use the web behavior, which is to allow the overrides, and to adjust the root container information, which is used for recording errors.

**Comment by tbitonti** [comment]
A candidate fix is building under the pull request #20016.

**Comment by mbroz2** [comment]
@tbitonti please update the description of this `relesae bug` using the [bug report template](https://github.com/OpenLiberty/open-liberty/issues/new?assignees=&labels=release+bug&template=bug_report.md&title=)
